### ISSUE TITLE
Why the power normalization need to plus 2?

### ISSUE THREAD

**Issue by wsxtyrdd** [mingyuyng/Deep-JSCC-for-images-with-OFDM#1]
on network.py line 29. 
class Normalize(nn.Module):
  def forward(self, x, power):
    pwr = torch.mean(x**2, (-2,-1), True) * 2
    return np.sqrt(power)*x/torch.sqrt(pwr)  

What is the meaning of "* 2"? It seems like the average power constraint is less than power/2. 
It confuses me.

**Comment by mingyuyng** [comment]
Hi @wsxtyrdd. When we do torch.mean(x**2, (-2,-1), True), we are essentially getting half of the signal power because the mean is taken over both the real part and imaginary part. So we multiple it by 2 to get the true power. Hope this makes sense to you.

**Comment by WenBingo** [comment]
> on network.py line 29. class Normalize(nn.Module): def forward(self, x, power): pwr = torch.mean(x**2, (-2,-1), True) * 2 return np.sqrt(power)*x/torch.sqrt(pwr)
> 
> What is the meaning of "* 2"? It seems like the average power constraint is less than power/2. It confuses me.

hello, can I ask your torch version number?  The torch.ifft() report an error, even when I change it to torch.fft.ifft() also report an error
